rclcpp/include/rclcpp/serialization.hpp

```diff
@@ -0,0 +1,158 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP__SERIALIZATION_HPP_
+#define RCLCPP__SERIALIZATION_HPP_
+
+#include <memory>
+#include <string>
+
+#include "rosidl_typesupport_cpp/message_type_support.hpp"
+
+#include "rcl/types.h"
+#include "rosidl_runtime_c/message_type_support_struct.h"
+
+namespace rclcpp
+{
+
+class SerializedMessage;
+
+/// Interface to (de)serialize a message
+class SerializationBase
+{
+protected:
```

Comment:
I just realized that this class is protected.

I don't think that's right here. Essentially, this would disable the need we'd have for this within ROSbags to avoid having a compile dependency for the message types.